access-control-app/src/cmr/access_control/services/acl_util.clj

```diff
@@ -102,3 +103,30 @@
                             (qm/string-condition :target target))
                     identity-type-condition)]
     (get-acls-by-condition context condition)))
+
+(defn sync-entry-titles-concept-ids
+  "If the given ACL is a catalog item acl with a collection identifier that includes concept-ids or
+   entry-titles, return ACL such that both are a unioned with each other for :create action. If action is
+   :update and there are only entry-titles, then sync against entry-titles, otherwise sync against concept-ids.
+   If there are no concept-ids or entry-titles, acl remains unchanged."
+  [context action acl]
+  (if-let [collection-identifier (get-in acl [:catalog-item-identity :collection-identifier])]
+    (let [entry-titles (:entry-titles collection-identifier)
+          concept-ids (:concept-ids collection-identifier)
+          provider-id (get-in acl [:catalog-item-identity :provider-id])
+          colls-from-entry-titles (when (and (seq entry-titles)
```

Comment:
Why not `(when (seq entry-titles) ...`?

Basically this logic is saying to ignore entry titles when there is an update that included both concept-ids and entry titles.

As an API user this would be really confusing to me that you silently ignored my entry titles. If we don't want both entry-titles and concept-ids to come in an update request we should add a validation and reject the request instead.